Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1884,15 +1885,37 @@ class BrowserViewController: UIViewController, SearchBarLocationProvider, Themea
 
                 tabWebView.accessoryView.savedCardsClosure = {
                     DispatchQueue.main.async { [weak self] in
-                        self?.showBottomSheetCardViewController(creditCard: nil,
-                                                                decryptedCard: nil,
-                                                                viewType: .selectSavedCard)
+                        // Dismiss keyboard
+                        webView.resignFirstResponder()
+                        // Authenticate and show bottom sheet with select a card flow
+                        self?.authenticateSavedCreditCardBottomSheet(fieldValues: fieldValues)
                     }
                 }
             }
         }
     }
 
+    private func authenticateSavedCreditCardBottomSheet(fieldValues: UnencryptedCreditCardFields) {
+        guard let appAuthenticator else {
+            return
+        }
+        appAuthenticator.getAuthenticationState { [unowned self] state in
+            switch state {
+            case .deviceOwnerAuthenticated:
+                self.showBottomSheetCardViewController(creditCard: nil,
+                                                       decryptedCard: nil,
+                                                       viewType: .selectSavedCard)
+            case .deviceOwnerFailed:
+                break // Keep showing bvc
+            case .passCodeRequired:
+                let passcodeViewController = DevicePasscodeRequiredViewController()
+                passcodeViewController.profile = self.profile
+                self.navigationController?.pushViewController(passcodeViewController,
+                                                              animated: true)
```

Comment:
FYI: This can be moved to use coordinators as we have done so in the settings area